src/test/java/Clase1/PracticoClase1.java

```diff
@@ -0,0 +1,367 @@
+
+            for (WebElement eachList : listH3){
+                System.out.println("H3 ---->" + eachList.getText());
+            }
+       }
+    }
+
+
+
+    @Test
+    public void ejercicioBusquedasAnidadas2(){
+
+        driver.get("https://www.apple.com");
+
+        List<WebElement> divList = driver.findElements(By.tagName("div"));
+        Assert.assertFalse(divList.isEmpty(), "La lista de DIV esta vacía");
+
+        int divsWithId = 0;
+
+        for (WebElement eachdiv : divList){
+
+            if (eachdiv.getAttribute("id") != ""){
+                System.out.println(eachdiv.getAttribute("id"));
+                divsWithId ++;
+            }
+        }
+
+        System.out.println("La cantidad de div con ID es" + " " + divsWithId);
+    }
+
+    @Test
+    public void ejercicioBusquedasAnidadas3(){
+        int divCount= 1;
+
+        driver.get("https://www.bbc.com");
+
+        List<WebElement> divList = driver.findElements(By.tagName("div"));
+        Assert.assertFalse(divList.isEmpty(), "La lista de DIV esta vacía");
+
+        for (WebElement eachDiv : divList){
+
+            List<WebElement> paragraphList = eachDiv.findElements(By.tagName("p"));
+            System.out.println("Div number " + divCount + " " + eachDiv.getAttribute("id"));
+
+
+            int paragraphCount = 1;
+            for (WebElement eachParagraph : paragraphList){
+                System.out.println("In div " +divCount +" Paragraph " + paragraphCount + " Texto: " + eachParagraph.getText());
+                paragraphCount++;
+            }
+            divCount++;
+        }
+    }
+
+    @Test
+    public void ejercicioGetAttribute1(){
+
+        driver.get("https://mundomac.com");
+
+        List<WebElement> listImg = driver.findElements(By.tagName("img"));
+        Assert.assertFalse(listImg.isEmpty(), "La lista esta vacia");
+        int countAltWithOutText = 0;
+        int countAltWithText = 0;
+        for(WebElement eachImg : listImg){
+
+            if (eachImg.getAttribute("alt") != "" ){
+                System.out.println(eachImg.getAttribute("alt"));
+                countAltWithText++;
+                //eachImg.getAttribute("alt") == ""
+                //eachImg.getAttribute("alt").isEmpty()== true
+
+            }else { countAltWithOutText++;}
+
+        }
+        System.out.println("La cantidad de IMG con texto son "+ countAltWithText);
+        System.out.println("La cantidad de IMG sin texto alternativo son " + countAltWithOutText);
+        //driver.close();
+    }
+
+    @Test
+    public void ejercicioGetAttribute2(){
+
+        driver.get("https://mundomac.com");
+        boolean sameClassName = true;
+        List<WebElement> listBtn = driver.findElements(By.tagName("button"));
+        Assert.assertFalse(listBtn.isEmpty(), "La lista de botones esta vacía");
+
+        for (WebElement eachBtn : listBtn){
+
+            if(eachBtn.getAttribute("class") == "btn btn-default"){
```

Comment:
este if que haria?

Reply:
Este IF lo que hace es comparar que para cada botón tenga la misma clase.  En caso de ser distinta la clase corta el for y pone la bandera en false. Con eso falla el Assert y avisa que no todos los botones usan la misma clase.